toolchain/testing/yaml_test_helpers.h

```diff
@@ -58,6 +58,19 @@
 
 namespace Carbon::Testing::Yaml {
 
+// Adds the specified indentation before each newline in the given string.
+inline auto IndentString(std::string_view str, std::string_view indent = "  ")
+    -> std::string {
+  std::string result;
+  for (auto c : str) {
+    result.push_back(c);
+    if (c == '\n') {
+      result += indent;
+    }
+  }
+  return result;
```

Comment:
```suggestion
  return absl::StrReplaceAll(str, {{"\n", "\n  "}});
```

I was considering suggesting `result.reserve(str.size())`, but maybe you can just include `absl/strings/str_replace.h` and let it handle the complexities? (also probably doing better than a char-by-char loop)